fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/batch/PushDownNotSlotReferenceExpressionOfOnClauseJob.java

```diff
@@ -0,0 +1,41 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.jobs.batch;
+
+import org.apache.doris.nereids.CascadesContext;
+import org.apache.doris.nereids.rules.rewrite.logical.PushDownNotSlotReferenceExpressionOfOnClause;
+
+import com.google.common.collect.ImmutableList;
+
+/**
+ * push down not slot reference expression job
+ */
+public class PushDownNotSlotReferenceExpressionOfOnClauseJob extends BatchRulesJob {
```

Comment:
The class name is not quite meaningful.  Here we want to push predicates like `t1.A=1`, but there is a SlotReference `t1.A`.
How about "SingleSidePredicate" since all slots comes from one side of this join.

Reply:
Yes, the name is pretty good!